src/lib/crypto/hash.cpp

```diff
@@ -190,14 +189,15 @@ pgp_hash_copy(pgp_hash_t *dst, const pgp_hash_t *src)
         return false;
     }
 
-    botan_hash_t handle;
-    if (botan_hash_copy_state(&handle, (botan_hash_t) src->handle)) {
+    Botan::HashFunction *hash_fn = static_cast<Botan::HashFunction*>(src->handle);
```

Comment:
What happens when `src->handle` is NULL? Do you think we should check for this case or just crash (which is sometimes fine)?